tests/project/test_maintenance.py

```diff
@@ -0,0 +1,81 @@
+import numpy as np
+from pyiron_base._tests import TestWithFilledProject
+from pyiron_base import GenericJob
+
+
+def _test_array(start=0):
+    return np.arange(start, start+100, dtype=object).reshape(5, 20)
+
+
+class TestMaintenance(TestWithFilledProject):
+
+    def setUp(self) -> None:
+        super().setUp()
+        job: GenericJob = self.project['toy_1']
+        job['user/some'] = _test_array(5)
+        job['user/some'] = _test_array()
+        self.initial_toy_1_hdf_file_size = job.project_hdf5.file_size(job.project_hdf5)
+        self.addTypeEqualityFunc(np.ndarray, self.assertArrayEqual)
+
+    def assertArrayEqual(self, first, second, msg=None):
+        self.assertIsInstance(first, np.ndarray, 'First argument is not a numpy array')
+        self.assertIsInstance(second, np.ndarray, 'Second argument is not a numpy array')
+
+        if not np.array_equal(first, second):
+            if msg is None:
+                self.fail(f'Arrays differ in shape or value {first} and {second}.')
+            else:
+                self.fail(msg)
```

Comment:
There's also [this nice utility](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_array_equal.html) from numpy itself.  I think we also came up with something like this a few times now,  I'll have a look at the other code quickly.